src/dcd/server/autocomplete/ufcs.d

```diff
@@ -132,6 +129,14 @@ bool willImplicitBeUpcasted(string from, string to)
     return INTEGER_PROMOTIONS[from] == to;
 }
 
+bool matchAliasThis(const(DSymbol)* aliasThisSymbol, const(DSymbol)* incomingSymbol)
+{
+    if(aliasThisSymbol) {
+        return isCallableWithArg(incomingSymbol, aliasThisSymbol.type);
```

Comment:
we should probably add a recursion check parameter, I think `alias this` to itself could crash the server through this right now.

Reply:
Ah right!

Reply:
want to do this?

Reply:
It should be done in my commit 
```
if (!beforeDotType.aliasThisSymbols
        || !beforeDotType.aliasThisSymbols.front
        || beforeDotType.aliasThisSymbols.front == beforeDotType)
    {
        return false;
    }
```

Reply:
this doesn't cover this case:
```d
struct A {
	B b;

	alias b this;
}

struct B {
	A a;

	alias a this;
}
```

invalid semantically, but could still be written and crash DCD. Consider just adding a maximum depth to avoid other cases we didn't think about.

Reply:
I understand now thanks